x/leverage/keeper/iter.go

```diff
@@ -45,14 +47,22 @@ func (k Keeper) GetAllRegisteredTokens(ctx sdk.Context) []types.Token {
 // GetAllSpecialAssetPairs returns all the special asset pairs from the x/leverage
 // module's KVStore.
 func (k Keeper) GetAllSpecialAssetPairs(ctx sdk.Context) []types.SpecialAssetPair {
-	return store.MustLoadAll[*types.SpecialAssetPair](ctx.KVStore(k.storeKey), types.KeyPrefixSpecialAssetPair)
+	pairs := store.MustLoadAll[*types.SpecialAssetPair](ctx.KVStore(k.storeKey), types.KeyPrefixSpecialAssetPair)
+	sort.SliceStable(pairs, func(i, j int) bool {
+		return pairs[i].CollateralWeight.LT(pairs[j].CollateralWeight)
+	})
+	return pairs
 }
 
 // GetSpecialAssetPairs returns all the special asset pairs from the x/leverage
 // module's KVStore which match a single asset.
 func (k Keeper) GetSpecialAssetPairs(ctx sdk.Context, denom string) []types.SpecialAssetPair {
 	prefix := types.KeySpecialAssetPairOneDenom(denom)
```

Comment:
This is why we needed directional store - it is now possible to load only relevant special asset pairs instead of all of them (as sets don't get the denom prefix)